src/transformers/trainer.py

```diff
@@ -5100,9 +5097,10 @@ def get_batch_samples(self, epoch_iterator, num_batches):
         if len(batch_samples) > 0 and "labels" in batch_samples[0]:
             # For now we don't support object detection
             try:
-                num_items_in_batch = sum(
-                    [data_batch["labels"][..., 1:].ne(-100).sum().item() for data_batch in batch_samples]
-                )
+                num_items_in_batch = sum([(batch["labels"].ne(-100)).sum() for batch in batch_samples])
```

Comment:
weird to me that we have to use -100 here, instead of a general parameter but whit was already the case

Reply:
IIRC we use `-100` for padding by default in the Trainer. I can align it to `self.processor` if it exists else `-100` if that's better?:)

Reply:
okay sounds good then sorry

Reply:
No worries, it's weird for me too :)

Reply:
Why do we no longer need to shift labels `["labels"][...,1:]`  when getting num_items_in_batch?